src/MacVim/MMFullScreenWindow.m

```diff
@@ -104,7 +104,17 @@ - (MMFullScreenWindow *)initWithWindow:(NSWindow *)t view:(MMVimView *)v
 
     // NOTE: Vim needs to process mouse moved events, so enable them here.
     [self setAcceptsMouseMovedEvents:YES];
-
+  
+    fadeTime = [[NSUserDefaults standardUserDefaults] doubleForKey:MMFullScreenFadeTimeKey];
+
+    // Each fade goes in and then out, so the fade hardware must be reserved accordingly and the
+    // actual fade time can't exceed half the allowable reservation time... plus some slack to
+    // prevent visual artifacts caused by defaulting on the fade hardware lease.
+    if (fadeTime > 0.45 * kCGMaxDisplayReservationInterval) {
```

Comment:
fadeTime = MAX(fadeTime, 0.45 * kCGMaxDisplayReservationInterval);

like https://github.com/macvim-dev/macvim/blob/master/src/gui_gtk_x11.c#L5083? It might need extra include though.